plugins/DevicesDetection/Columns/DeviceBrand.php

```diff
@@ -60,4 +60,15 @@ public function onNewVisit(Request $request, Visitor $visitor, $action)
 
         return $parser->getBrand();
     }
+
+    /**
+     * @param Request $request
+     * @param Visitor $visitor
+     * @param Action|null $action
+     * @return mixed
+     */
+    public function onAnyGoalConversion(Request $request, Visitor $visitor, $action)
```

Comment:
I think it would be better to recreate PR against 3.x because of the schema change. Otherwise looks 👍